app/services/caldav.py

```
@@ -0,0 +1,46 @@
+
+        headers = {
+            'Content-Type': 'text/calendar; charset=utf-8',
+            'User-Agent': 'May-Vehicle-Manager/1.0',
+        }
+        if username and password:
+            token = b64encode(f'{username}:{password}'.encode('utf-8')).decode('ascii')
+            headers['Authorization'] = f'Basic {token}'
+        if event.external_etag:
+            headers['If-Match'] = event.external_etag
+
+        request = Request(event_url, data=ics.encode('utf-8'), headers=headers, method='PUT')
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)